### PR TITLE
Merge release apimanagement 2023 05 01 preview

### PR DESCRIPTION
**Pull request by solankisamir**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by TimLovellSmith** [comment]
Removing MergeRequested because this PR doesn't seem to satisfy merge requirements yet.

**Comment by TimLovellSmith** [comment]
Please check the lintdiff errors:

<html>
<body>




❌ LroErrorContent | Error response content of long running operations must follow the error schema provided in the common types v2 and above.Location: Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json#L478 |  
❌ LroErrorContent | Error response content of long running operations must follow the error schema provided in the common types v2 and above.Location: Microsoft.ApiManagement/preview/2023-05-01-preview/apimusers.json#L407 |  
⚠️ PostOperationIdContainsUrlVerb | OperationId should contain the verb: 'validatepolicies' in:'PolicyRestrictionValidations_ByService'. Consider updating the operationIdLocation: Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L42 |  
⚠️ LatestVersionOfCommonTypesMustBeUsed | Use the latest version v5 of types.json.Location: Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L82

<br class="Apple-interchange-newline">
</body>
</html>❌ [LroErrorContent](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-error-content.md)	Error response content of long running operations must follow the error schema provided in the common types v2 and above.
Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json#L478](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json#L478)	
❌ [LroErrorContent](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-error-content.md)	Error response content of long running operations must follow the error schema provided in the common types v2 and above.
Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimusers.json#L407](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimusers.json#L407)	
⚠️ [PostOperationIdContainsUrlVerb](https://github.com/Azure/azure-openapi-validator/blob/main/docs/post-operation-id-contains-url-verb.md)	OperationId should contain the verb: 'validatepolicies' in:'PolicyRestrictionValidations_ByService'. Consider updating the operationId
Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L42](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L42)	
⚠️ [LatestVersionOfCommonTypesMustBeUsed](https://github.com/Azure/azure-openapi-validator/blob/main/docs/latest-version-of-common-types-must-be-used.md)	Use the latest version v5 of types.json.
Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L82](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L82)

**Comment by solankisamir** [comment]
> Please check the lintdiff errors:
> 
> ❌ LroErrorContent | Error response content of long running operations must follow the error schema provided in the common types v2 and above.Location: Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json#L478 |   ❌ LroErrorContent | Error response content of long running operations must follow the error schema provided in the common types v2 and above.Location: Microsoft.ApiManagement/preview/2023-05-01-preview/apimusers.json#L407 |   ⚠️ PostOperationIdContainsUrlVerb | OperationId should contain the verb: 'validatepolicies' in:'PolicyRestrictionValidations_ByService'. Consider updating the operationIdLocation: Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L42 |   ⚠️ LatestVersionOfCommonTypesMustBeUsed | Use the latest version v5 of types.json.Location: Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L82
> 
> ❌ [LroErrorContent](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-error-content.md) Error response content of long running operations must follow the error schema provided in the common types v2 and above. Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json#L478](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json#L478) ❌ [LroErrorContent](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-error-content.md) Error response content of long running operations must follow the error schema provided in the common types v2 and above. Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimusers.json#L407](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimusers.json#L407) ⚠️ [PostOperationIdContainsUrlVerb](https://github.com/Azure/azure-openapi-validator/blob/main/docs/post-operation-id-contains-url-verb.md) OperationId should contain the verb: 'validatepolicies' in:'PolicyRestrictionValidations_ByService'. Consider updating the operationId Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L42](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L42) ⚠️ [LatestVersionOfCommonTypesMustBeUsed](https://github.com/Azure/azure-openapi-validator/blob/main/docs/latest-version-of-common-types-must-be-used.md) Use the latest version v5 of types.json. Location: [Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L82](https://github.com/Azure/azure-rest-api-specs/blob/4d4bef5d4a31f3e480f6b7bfa4b419298fd0b285/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json#L82)

Hey @TimLovellSmith this change is coming from https://github.com/Azure/azure-rest-api-specs/pull/24696 which was ARM signed off

Essentially this contract existed previous also and if we do this change now, it will be considered as a breaking change. As far as I understand we need to update schema to common types only for new entities, otherwise it triggers a breaking change for some clients.

**Comment by TimLovellSmith** [comment]
@rkmanda Should the linter errors be addressed now, suppressed permanently, or addressed in a later PR?

File: custom-words.txt [hunk 1]
```diff
@@ -3931,4 +3931,6 @@ capacityreservationgroup
 capacityreservationgroups
 ciam
 CIAM
+protobuf
+proto
```
Comment (TimLovellSmith) [inline custom-words.txt:3935]: do you actually need 'proto' or only 'protobuf'?

**Comment by TimLovellSmith** [comment]
>               },

its not typical to return location header with 200

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json:340 in 1955ddf. [](commit_id = 1955ddfdc63a6c1bc2888b26b10657bf364682d2, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>               "location": {

its not typical to return location header with 201 and 200

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json:3048 in 1955ddf. [](commit_id = 1955ddfdc63a6c1bc2888b26b10657bf364682d2, deletion_comment = False)

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimnamedvalues.json [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Request to refresh secret was accepted.",
+            "headers": {
+              "location": {
```
Comment (TimLovellSmith) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimnamedvalues.json:510]: >location [](http://example.com/codeflow?start=15&length=8)

Isn't location header supposed to be capitalized 'Location'?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimnamedvalues.json [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+            "description": "Named value was successfully created.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/NamedValueContract"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              },
+              "location": {
```
Comment (TimLovellSmith) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimnamedvalues.json:259]: >location [](http://example.com/codeflow?start=15&length=8)

Isn't location header supposed to be capitalized 'Location'?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "location": {
```
Comment (TimLovellSmith) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimpolicyrestrictionsvalidation.json:67]: >location [](http://example.com/codeflow?start=15&length=8)

Isn't location header supposed to be capitalized 'Location'?

**Comment by TimLovellSmith** [comment]
>         "location": "https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/apis/tempgroup?api-version=2023-05-01-preview&asyncId=5c730e343244df1b9cb56e85&asyncCode=201",

shouldn't it be capitalized Location?

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/examples/ApiManagementCreateApi.json:34 in 1955ddf. [](commit_id = 1955ddfdc63a6c1bc2888b26b10657bf364682d2, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
@rkmanda @JeffreyRichter Some remaining worries, not sure they are blockers, seeking more opinions - is lowercase 'location' header okay? Is returning a 'location' header with 200 OK okay?

**Comment by JeffreyRichter** [comment]
JSON properties are always camelCase and case-sensitive.
I think this is a header though and headers are case-insensitive, but we do recommend all lowercase as HTTP/2 requires all lowercase so by doing this now, we are prepared for HTTP/2.

**Comment by TimLovellSmith** [comment]
I'm trying to find docs which say OK and location are fine together. 

This one seems to say the reverse, that they're mutually exclusive:
https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations#status-codes-for-asynchronous-operations

This one doesn't seem to have an opinion on Location headers in particular, only Operation-Location?:
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md

**Comment by kazrael2119** [comment]
please fix duplicate schemas: ErrorResponse ProxyResource Resource, maybe you reference multi version from common-type

**Comment by ms-zhenhua** [comment]
please fix errors in https://github.com/Azure/azure-rest-api-specs/pull/26547/checks?check_run_id=19657201806 before adding the `MergeRequested` label

**Comment by solankisamir** [comment]
@raych1(GO) @msyyc(Python) @qiaozha(JS) can you please take a look at the breaking changes?

File: specification/apimanagement/resource-manager/readme.typescript.md [hunk 5]
```diff
@@ -9,6 +9,8 @@ typescript:
   package-name: "@azure/arm-apimanagement"
   output-folder: "$(typescript-sdks-folder)/sdk/apimanagement/arm-apimanagement"
   generate-metadata: true
+  modelerfour: 
+    lenient-model-deduplication: true
```
Comment (kazrael2119) [inline specification/apimanagement/resource-manager/readme.typescript.md:13]: I don't think this is a good solution to fix duplicate, could you unify the reference versions from common-type?

Comment (Alancere) [reply specification/apimanagement/resource-manager/readme.typescript.md:13]: Same for readme.go.md, please

File: custom-words.txt (hunk 1 above)
Comment (tomkerkhove) [reply custom-words.txt:3935]: We have both AFAIK

File: specification/apimanagement/resource-manager/readme.typescript.md (hunk 5 above)
Comment (tomkerkhove) [reply specification/apimanagement/resource-manager/readme.typescript.md]: This is rolled back

File: custom-words.txt (hunk 1 above)
Comment (TimLovellSmith) [reply custom-words.txt:3935]: @tomkerkhove Why? I don't think its a real word. Its just a prefix. Please don't add it in custom-words. Consdier adding it in cSpell intead!

Comment (tomkerkhove) [reply custom-words.txt:3935]: It's a commonly used term in gRPC space

**Review by ms-henglu — APPROVED** [review]

**Pull request merged**